components/scream/src/control/surface_coupling.cpp

```diff
@@ -113,17 +141,124 @@ register_export (const std::string& fname,
 
   auto& info = m_scream_exports_host(m_num_exports);
 
-  // Set view data ptr
-  info.data = field.get_view().data();
+  // 2 cases for setting export info:
+  //   1. General case: fname describes a field in the FieldManager which has no parent.
+  //      In this case, gather info directly from the field.
+  //   2. Special cases
+  //      a. qv: fname describes a field in the FieldManager which has a parent.
+  //         In this case, we need to create a view and store and copy over data
+  //      b. fname == set_zero, which indicates that this field is not
+  //         used in SCREAM. In this case, create a view with all entries
+  //         zero.
+  //      c. fname corresponeds to a field which is a combination of
+  //         SCREAM fields. In this case, query the FieldManager for
+  //         these fields and create fill the correct view with the computed
+  //         values (only at the surface).
+  if (m_field_mgr->has_field(fname)
+      &&
+      m_field_mgr->get_field(fname).get_header().get_parent().lock() == nullptr) {
 
-  // Set cpl index
-  info.cpl_idx = cpl_idx;
+    // If fname is a field in the the field manager, then set the data and column info based on this field.
+    // Note that the field must not have a parent as get_view() would be unsafe.
+    export_field_type field = m_field_mgr->get_field(fname);
+
+    EKAT_REQUIRE_MSG (field.is_allocated(), "Error! Export field view has not been allocated yet.\n");
+
+    // Set view data ptr
+    info.data = field.get_view().data();
+
+    // Get column offset and stride
+    get_col_info (field.get_header_ptr(), vecComp, info.col_offset, info.col_stride);
+
+    // Store the identifier of this field, for debug purposes
+    m_exports_fids.insert(field.get_header().get_identifier());
+
+  } else {
+
+    using PF = PhysicsFunctions<device_type>;
+    const int last_entry = m_num_levs-1;
 
-  // Get column offset and stride
-  get_col_info (field.get_header_ptr(), vecComp, info.col_offset, info.col_stride);
+    if (fname == "qv") {
 
-  // Store the identifier of this field, for debug purposes
-  m_exports_fids.insert(field.get_header().get_identifier());
+      const auto qv = m_field_mgr->get_field(fname).get_reshaped_view<const Real**>();
+      const auto policy = KokkosTypes<device_type>::RangePolicy(0, m_num_cols);
+      Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const int& i) {
```

Comment:
Why do you copy data from the FM field into qv_sfc at registration time? I would expect this operation to be done every time that do_export is called...